### PR TITLE
[Docs] Consolidate transactions documentation

### PR DESCRIPTION
**Pull request by movekevin**
### Description

Currently both Concepts > Transactions and States and Build Apps > System Integration Guide have a lot of details on transactions. This PR:
1. Consolidate transaction details under Concepts > Transactions and States
2. Reorder the page so that States come after Transactions. This is logical as States are generated from executing Transactions.

### CODE REVIEW

File: developer-docs-site/docs/concepts/basics-txns-states.md [hunk 1]
```diff
@@ -25,33 +25,12 @@ sources={{
   }}
 />
 
-## Ledger state
```
Comment (movekevin) [inline developer-docs-site/docs/concepts/basics-txns-states.md:28]: Moved toward the end of the doc, after Transactions

File: developer-docs-site/docs/guides/system-integrators-guide.md [hunk 2]
```diff
@@ -61,135 +61,6 @@ Aptos currently provides three SDKs:
 * [REST API spec](https://fullnode.devnet.aptoslabs.com/v1/spec#/)
 * [Local testnet development flow](/guides/local-testnet-dev-flow)
 
-## Accounts on Aptos
```
Comment (movekevin) [inline developer-docs-site/docs/guides/system-integrators-guide.md:64]: Consolidated under Concepts > Accounts (in a different PR that's already been merged)

File: developer-docs-site/docs/guides/system-integrators-guide.md [hunk 3]
```diff
@@ -61,135 +61,6 @@ Aptos currently provides three SDKs:
...
-
-### Preventing replay attacks
-
-When the Aptos blockchain processes the transaction, it looks at the sequence number in the transaction and compares it with the sequence number in the sender’s account (as stored on the blockchain at the current ledger version). The transaction is executed only if the sequence number in the transaction is the same as the sequence number for the sender account, and the transaction is rejected if those two numbers do not match. In this way past transactions, which necessarily contain older sequence numbers, cannot be replayed, hence preventing replay attacks.
-
-:::tip Read more
-See more on [Account sequence numbering](/concepts/basics-accounts#account-sequence-number).
-:::
-
-## Transactions
```
Comment (movekevin) [inline developer-docs-site/docs/guides/system-integrators-guide.md:102]: Consolidated under Concepts > Transactions and States in this PR.

**Review by clay-aptos — APPROVED** [review]
Please make the links relative.

**Comment by davidiw** [comment]
This PR left a whole. Now system integrators guide has NOTHING about transaction types. System integrators guide was intended to be an index into all content that eventually would push content out of it but still remain as an index.

**Pull request merged**